### PR TITLE
allow ENV as wavetable modulation source

### PR DESCRIPTION
**Pull request by friedolino78**

### CODE REVIEW

File: src/Params/ADnoteParameters.cpp [hunk 1]
```diff
@@ -1643,6 +1662,14 @@ void ADnoteVoiceParam::getfromXML(XMLwrapper& xml, unsigned nvoice)
             xml.exitbranch();
         }
 
+        PFMAmpEnvelopeEnabled = xml.getparbool("wave_envelope_enabled",
```
Comment (JohannesLorenz) [inline src/Params/ADnoteParameters.cpp]: `PFMAmpEnvelopeEnabled` ?

File: src/Synth/Envelope.cpp [hunk 2]
```diff
@@ -109,28 +109,25 @@ void Envelope::watch(float time, float value)
     float factor2;
     pos[0] = time;
     switch(mode) {
-        case 2:
```
Comment (JohannesLorenz) [inline src/Synth/Envelope.cpp]: The changes in this file are a cleanup which is independent from the WT feature, correct? In that case, could you please split it into another PR towards master, if it's not too difficult for you? I ask because the WT branch is already extremely large...

File: src/Params/ADnoteParameters.cpp [hunk 3]
```diff
@@ -49,6 +49,7 @@ static const Ports voicePorts = {
     rRecurp(FilterEnvelope, "Filter Envelope"),
     rRecurp(FMFreqEnvelope, "Modulator Frequency Envelope"),
     rRecurp(FMAmpEnvelope,  "Modulator Amplitude Envelope"),
+    rRecurp(WaveEnvelope,  "Wavetable Modulation Envelope"),
```
Comment (JohannesLorenz) [inline src/Params/ADnoteParameters.cpp]: Can you please move the string `"Wavetable Modulation Envelope` one space to the right, so the alignment is correct?

File: src/Synth/Envelope.cpp (hunk 2 above)
Comment (friedolino78) [reply src/Synth/Envelope.cpp]: ok

File: src/Params/ADnoteParameters.cpp (hunk 1 above)
Comment (friedolino78) [reply src/Params/ADnoteParameters.cpp]: right

File: src/Params/ADnoteParameters.cpp (hunk 3 above)
Comment (friedolino78) [reply src/Params/ADnoteParameters.cpp]: ok

File: src/Synth/ADnote.cpp [hunk 4]
```diff
@@ -1794,104 +1823,108 @@ inline void ADnote::ComputeVoiceOscillatorFrequencyModulation(int nvoice,
...
 
-            for(i = 0; i < synth.buffersize; ++i) {
-                tw[i] =
-                    (NoteVoicePar[nvoice].FMSmp[poshiFM] * (1.0f - posloFM)
-                     + NoteVoicePar[nvoice].FMSmp[poshiFM + 1] * posloFM);
-                posloFM += freqloFM;
-                if(posloFM >= 1.0f) {
-                    posloFM = fmod(posloFM, 1.0f);
-                    poshiFM++;
+            for(int i = 0; i < synth.buffersize; ++i) {
```
Comment (JohannesLorenz) [inline src/Synth/ADnote.cpp]: Is this a merge error that this fell into another line? Or is it wanted?

**Review by JohannesLorenz — COMMENTED** [review]
Only 1 comment :+1:

File: src/Synth/ADnote.cpp (hunk 4 above)
Comment (friedolino78) [reply src/Synth/ADnote.cpp]: indeed a merge error. now fixed.

File: src/Synth/ADnote.cpp [hunk 5]
```diff
@@ -1896,7 +1895,7 @@ inline void ADnote::ComputeVoiceOscillatorWaveTableModulation(int nvoice, FMTYPE
 
     //do the modulation
     for(int k = 0; k < vce.unison_size; ++k) {
-        float *smps   = NoteVoicePar[nvoice].OscilSmp.smps;
+        //~ float *smps   = NoteVoicePar[nvoice].OscilSmp.smps;
```
Comment (JohannesLorenz) [inline src/Synth/ADnote.cpp]: That can be removed, right?

File: src/Synth/ADnote.cpp [hunk 6]
```diff
@@ -1796,6 +1824,9 @@ inline void ADnote::ComputeVoiceOscillatorFrequencyModulation(int nvoice,
  */
 inline void ADnote::ComputeVoiceOscillatorWaveTableModulation(int nvoice, FMTYPE FMmode)
 {
+  Voice& vce = NoteVoicePar[nvoice];
```
Comment (JohannesLorenz) [inline src/Synth/ADnote.cpp]: Can you fix the indentation here, please?

File: src/Synth/ADnote.cpp (hunk 5 above)
Comment (friedolino78) [reply src/Synth/ADnote.cpp]: yes

File: src/Synth/ADnote.cpp (hunk 6 above)
Comment (friedolino78) [reply src/Synth/ADnote.cpp]: done

**Comment by JohannesLorenz** [comment]
@friedolino78 Thanks a lot for the PR! :tada:
